### PR TITLE
fix: config perm

### PR DESCRIPTION
**Pull request by AkashKumar7902**
## Related Issue
  - Info about Issue or bug

Closes: #1145 

#### Describe the changes you've made
added chmod function to change permission for keploy-config file

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by PranshuSrivastava — APPROVED** [review]
LGTM!

File: pkg/service/generateConfig/generateConfig.go [hunk 1]
```diff
@@ -108,5 +109,12 @@ func (g *generatorConfig) GenerateConfig(filePath string) {
 		g.logger.Fatal("Failed to write config file", zap.Error(err))
 	}
 
+
+  cmd := exec.Command("sudo", "chmod", "-R", "777", filePath)
```
Comment (charankamarapu) [inline pkg/service/generateConfig/generateConfig.go:113]: `err = os.WriteFile(filePath, results, os.ModePerm)` here os.ModePerm should give the read and write permissions. Firstly Check why it is not working properly or if there is any other permission mode which enables add it.

File: pkg/service/generateConfig/generateConfig.go [hunk 2]
```diff
@@ -108,5 +109,12 @@ func (g *generatorConfig) GenerateConfig(filePath string) {
...
+  err = cmd.Run()
+  if err != nil {
+    g.logger.Error("failed to set the permission of config file", zap.Error(err))
```
Comment (charankamarapu) [inline pkg/service/generateConfig/generateConfig.go:116]: if you don't return here , the logger which is below will also get executed.

**Review by charankamarapu — COMMENTED** [review]
Please correct them in any other PR.

File: pkg/service/generateConfig/generateConfig.go (hunk 1 above)
Comment (AkashKumar7902) [reply pkg/service/generateConfig/generateConfig.go:113]: is this clear ?

Comment (charankamarapu) [reply pkg/service/generateConfig/generateConfig.go:113]: yep

**Pull request merged**